epaint/src/text/text_layout.rs

```diff
@@ -194,8 +194,8 @@ fn line_break(
     let mut row_start_idx = 0;
     let mut non_empty_rows = 0;
 
-    for (i, glyph) in paragraph.glyphs.iter().enumerate() {
-        let potential_row_width = glyph.max_x() - row_start_x;
+    for (i, glyphs) in (0..paragraph.glyphs.len()).map(|i| &paragraph.glyphs[i..]).enumerate() {
+        let potential_row_width = glyphs[0].max_x() - row_start_x;
```

Comment:
This is a bit of a mouthful. I would suggest something simpler:

```suggestion
    for i in ..paragraph.glyphs.len() {
        let potential_row_width = paragraph.glyphs[i].max_x() - row_start_x;
```

And then use `row_break_candidates.add(i, &paragraph.glyphs[i..]);` later

Reply:
Done, thanks!